Add section about StarCoder for Enterprise (#60)
sourcegraph/docs: docs/cody/clients/enable-cody-enterprise.mdx

/>

## Use StarCoder for Autocomplete

> In our testing with other coder models for the autocomplete use case, [StarCoder](https://huggingface.co/blog/starcoder) offered significant improvements in quality and latency compared to our control groups for users on Sourcegraph.com. You can read more about the improvements in our [October 2023 release notes](https://sourcegraph.com/blog/feature-release-october-2023) and the [GA release notes](https://sourcegraph.com/blog/cody-is-generally-available).

Reviewer: ```suggestion
When tested with other coder models for the autocomplete use case, [StarCoder](https://huggingface.co/blog/starcoder) offered significant improvements in quality and latency compared to our control groups for users on Sourcegraph.com. You can read more about the improvements in our [October 2023 release notes](https://sourcegraph.com/blog/feature-release-october-2023) and the [GA release notes](https://sourcegraph.com/blog/cody-is-generally-available).
```

> To be able to offer our Enterprise users the best experience for Cody Autocomplete, we have partnered with [Fireworks](https://fireworks.ai/) and have set up an enterprise-grade deployment with dedicated hardware capacity. Since Sourcegraph 5.3, every Cody setup that uses Cody Gateway is able to opt-in to using [StarCoder](https://huggingface.co/blog/starcoder) as an alternative to Claude Instant as the AI model for Autocomplete.

Reviewer: ```suggestion
To get the best Cody autocomplete experience, Enterprise users can set up an enterprise-grade deployment with dedicated hardware with [Fireworks](https://fireworks.ai/). Since Sourcegraph 5.3, every Cody setup that uses [Cody Gateway](/cody/core-concepts/cody-gateway) can opt-in to using [StarCoder](https://huggingface.co/blog/starcoder) as an alternative to Claude Instant as the AI model for autocomplete.
```

> To enable StarCoder go to **Site admin > Site configuration** (`/site-admin/configuration`) and change the `completionModel`:

Reviewer 2: @eshamow @michaellzc would we expect the Cloud customers to make this change manually? Or, would they ask us (Sourcegraph) to apply the change to a given Cloud instance?

```json
{
  // [...]
  "cody.enabled": true,
  "completions": {
    "provider": "sourcegraph",
    "completionModel": "fireworks/starcoder"
  }
}
```

> Clients will automatically pick up this change when they connect to your Enterprise instance.

Reviewer 2: ```suggestion
Cody Extension will automatically pick up this change when they connect to your Enterprise instance.
```

Cody Extension might be a more natural term for external audience?

> We highly recommend every instance admin to make this change and plan to make this the default in a future release.

Reviewer: ```suggestion
It's recommended that every instance admin makes this change and plans to make this the default in a future release.
```

> We highly recommend every instance admin to make this change and plan to make this the default in a future release.

Reviewer 3: ```suggestion
We highly recommend every instance admin to make this change and plan to make this the default in a future release, for anyone not using a third-party LLM provider.
```
